service/rtc/sfu.go

```diff
@@ -275,16 +297,26 @@ func (s *Server) InitSession(cfg SessionConfig, closeCb func() error) error {
 					return
 				}
 
-				rtp := &rtp.Packet{}
-				if err := rtp.Unmarshal(buf[:i]); err != nil {
+				packet := &rtp.Packet{}
+				if err := packet.Unmarshal(buf[:i]); err != nil {
 					s.log.Error("failed to unmarshal RTP packet",
 						mlog.Err(err), mlog.String("sessionID", us.cfg.SessionID))
 					s.metrics.IncRTCErrors(us.cfg.GroupID, "rtp")
 					return
 				}
 
+				if us.vadMonitor != nil && audioLevelExtensionID > 0 {
+					var ext rtp.AudioLevelExtension
+					audioExtData := packet.GetExtension(uint8(audioLevelExtensionID))
+					if err := ext.Unmarshal(audioExtData); err != nil {
+						s.log.Error("failed to unmarshal audio level extension",
+							mlog.Err(err), mlog.String("sessionID", us.cfg.SessionID))
+					}
+					us.vadMonitor.PushAudioLevel(ext.Level)
+				}
```

Comment:
This is really cool -- but what happens when you mute while in a voiceOn state? 

One possible solution? Maybe we need send a voiceOff on mute. It may be redundant sometimes, but I think the clients will gracefully handle it.

Reply:
Thanks. For some reason I thought we were being smart on the client and correctly updated the voice state upon receiving a `user_muted` event but it looks like we are actually relying on an explicit voice off event on mute.

I guess we could keep doing the same but that would likely require a small addition on mobile so maybe we do it on this side although ultimately it's a bit of a wasted event since the state could be correctly inferred on client alone.